### ISSUE TITLE
Crazy Taxi:  invalid display list index

### ISSUE THREAD

**Issue by snickerbockers** [washingtondc-emu/washingtondc#23]
Steps to reproduce: run Crazy Taxi in arcade mode with Axel as the driver.  WashingtonDC will crash with the dump pasted below during the intro.

This probably isn't tied to Axel or the arcade mode, that's just the only way I've tried to reproduce this bug.

This bug was introduced when I added an additional sanity check in 90be8a5b7b7c2e0127a33e8348f94910cf816586, although the root-cause is obviously older than that.

```
POLY HEADER PACKET!
WARNING: TA_CMD_TYPE_UNKNOWN received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_INPUT_LIST received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_UNKNOWN received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_INPUT_LIST received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_UNKNOWN received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_INPUT_LIST received on pvr2 ta fifo!
Total elapsed time: 767 seconds and 459132956 nanoseconds
8629609 SH4 CPU cycles executed
Performance is 16.800284 MHz (8.400142%)
ERROR: unable to continue due to unimplemented functionality
[sh4_reg_mmucr] = 0
[sh4_reg_tea] = 0
[sh4_reg_ttb] = 0
[sh4_reg_ptea] = 0
[sh4_reg_ptel] = 0
[sh4_reg_pteh] = 0
[sh4_reg_qacr1] = 10
[sh4_reg_qacr0] = 10
[sh4_reg_ccr] = 90d
[sh4_reg_r15] = c00f384
[sh4_reg_r14] = 0
[sh4_reg_r13] = 60
[sh4_reg_r12] = acea0000
[sh4_reg_r11] = c2bc900
[sh4_reg_r10] = 0
[sh4_reg_r9] = 0
[sh4_reg_r8] = f0ffffff
[sh4_reg_r7_bank1] = c2bc960
[sh4_reg_r6_bank1] = 2fb
[sh4_reg_r5_bank1] = 0
[sh4_reg_r4_bank1] = c2bc960
[sh4_reg_r3_bank1] = a05f6804
[sh4_reg_r2_bank1] = a05f6800
[sh4_reg_r1_bank1] = 1
[sh4_reg_r0_bank1] = a05f6808
[sh4_reg_r7_bank0] = 0
[sh4_reg_r6_bank0] = 0
[sh4_reg_r5_bank0] = 0
[sh4_reg_r4_bank0] = 0
[sh4_reg_r3_bank0] = 0
[sh4_reg_r2_bank0] = 0
[sh4_reg_r1_bank0] = 0
[sh4_reg_r0_bank0] = 0
[sh4_reg_fpul] = ff
[sh4_reg_fpscr] = 240001
[sh4_reg_pr] = c15822c
[sh4_reg_macl] = 68
[sh4_reg_mach] = 0
[sh4_reg_dbr] = 8c000010
[sh4_reg_sgr] = c00f3d0
[sh4_reg_vbr] = 8c00f400
[sh4_reg_gbr] = c2b5060
[sh4_reg_spc] = c15832a
[sh4_reg_pc] = c15340a
[sh4_reg_ssr] = 60000001
[sh4_reg_sr] = 600000e0
[function] = "on_polyhdr_received"
[file] = "/home/jay/programs/washingtondc/src/hw/pvr2/pvr2_ta.c"
[line] = 535
[display_list_index] = 7
[feature] = "correct response for invalid display list indices"
Aborted (core dumped)
```

**Comment by snickerbockers** [comment]
Here's a new dump including the TA fifo contents (which were added to the crash dumps for this error in c8a5f233184e0da3f41be189c84076f67a8759b3).

```
WARNING: TA_CMD_TYPE_UNKNOWN received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_INPUT_LIST received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_UNKNOWN received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_INPUT_LIST received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_UNKNOWN received on pvr2 ta fifo!
WARNING: TA_CMD_TYPE_INPUT_LIST received on pvr2 ta fifo!
Total elapsed time: 1226 seconds and 999604844 nanoseconds
1746355554 SH4 CPU cycles executed
Performance is 15.424801 MHz (7.712400%)
ERROR: unable to continue due to unimplemented functionality
[sh4_reg_mmucr] = 0
[sh4_reg_tea] = 0
[sh4_reg_ttb] = 0
[sh4_reg_ptea] = 0
[sh4_reg_ptel] = 0
[sh4_reg_pteh] = 0
[sh4_reg_qacr1] = 10
[sh4_reg_qacr0] = 10
[sh4_reg_ccr] = 90d
[sh4_reg_r15] = c00f384
[sh4_reg_r14] = 0
[sh4_reg_r13] = a0
[sh4_reg_r12] = acea0000
[sh4_reg_r11] = c2bc900
[sh4_reg_r10] = 0
[sh4_reg_r9] = 0
[sh4_reg_r8] = f0ffffff
[sh4_reg_r7_bank1] = c2bc9a0
[sh4_reg_r6_bank1] = 2fb
[sh4_reg_r5_bank1] = 0
[sh4_reg_r4_bank1] = c2bc9a0
[sh4_reg_r3_bank1] = a05f6804
[sh4_reg_r2_bank1] = a05f6800
[sh4_reg_r1_bank1] = 1
[sh4_reg_r0_bank1] = a05f6808
[sh4_reg_r7_bank0] = 0
[sh4_reg_r6_bank0] = 0
[sh4_reg_r5_bank0] = 0
[sh4_reg_r4_bank0] = 0
[sh4_reg_r3_bank0] = 0
[sh4_reg_r2_bank0] = 0
[sh4_reg_r1_bank0] = 0
[sh4_reg_r0_bank0] = 0
[sh4_reg_fpul] = ff
[sh4_reg_fpscr] = 240001
[sh4_reg_pr] = c15822c
[sh4_reg_macl] = 68
[sh4_reg_mach] = 0
[sh4_reg_dbr] = 8c000010
[sh4_reg_sgr] = c00f3d0
[sh4_reg_vbr] = 8c00f400
[sh4_reg_gbr] = c2b5060
[sh4_reg_spc] = c15832a
[sh4_reg_pc] = c15340a
[sh4_reg_ssr] = 60000001
[sh4_reg_sr] = 600000e0
[function] = "on_polyhdr_received"
[file] = "/home/jay/programs/washingtondc/src/hw/pvr2/pvr2_ta.c"
[line] = 569
[ta_fifo_word_f] = 3f800000
[ta_fifo_word_e] = 3f800000
[ta_fifo_word_d] = 3f800000
[ta_fifo_word_c] = 0
[ta_fifo_word_b] = 3e560419
[ta_fifo_word_a] = 3eb645a2
[ta_fifo_word_9] = 3f30624e
[ta_fifo_word_8] = 3f800000
[ta_fifo_word_7] = 4410cdfa
[ta_fifo_word_6] = 3c13e112
[ta_fifo_word_5] = 4398ed52
[ta_fifo_word_4] = 441d7c56
[ta_fifo_word_3] = 3cd3db6a
[ta_fifo_word_2] = c22acad8
[ta_fifo_word_1] = 3ceb7edc
[ta_fifo_word_0] = bfd34467
[ta_fifo_byte_count] = 32
[display_list_index] = 7
[feature] = "correct response for invalid display list indices"
```

**Comment by snickerbockers** [comment]
Fixed by 46e1cd0ec54ce172514a5e101d114c9f3ec47b6a.
